### PR TITLE
Move the Hive website to github pages.

### PR DESCRIPTION
**Pull request by omalley**
This pull request adds a site to our main branch that contains the source code to the site.

We can ask Infra to automatically build and deploy github-pages from that branch & directory.

To look at the translated site, there is a docker file and a readme that will build the site locally.

### CODE REVIEW

File: site/privacy_policy.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: Privary Policy
```
Comment (massdosage) [inline site/privacy_policy.md:2]:
```suggestion
title: Privacy Policy
```

**Comment by kgyrtkirk** [comment]
this patch adds more than 5000 files, mostly because of javadoc stuff...
I think javadoc is an implicit content - and as such I think it shouldn't be "part" of the project repository

**Comment by omalley** [comment]
Agreed that it is a lot of files, but if the project wants to continue to publish javadocs we need to host them somewhere. The current site keeps 6 versions in subversion. I cut that down to 1 in the conversion. I believe that Apache Infra wants to shut down the subversion that is currently hosting those javadocs.

I'm open to solutions, including stopping publishing Hive's javadocs. What would you suggest?

**Comment by kgyrtkirk** [comment]
I just wanted to express that I don't think it's the best approach to expose javadocs like this...I really appreciate your efforts migrating the site - a few options I can think of:
* push the site into a different branch; like `gh-pages` - so we can pospone to solve this later...
* we can stop publishing javadoc on the site for now...
* or add some download&unpack to the site deploy script - not sure if this is possible..will need to check this

**Comment by omalley** [comment]
One of the advantages of including the site in docs on the main branch is that it will automatically get packaged as part of the releases, so I'd rather not give that up.

How about if we put the generated javadoc in a branch named "javadoc-3.1". We could point the site to reference that branch on github. Would that be acceptable?

**Comment by kgyrtkirk** [comment]
> One of the advantages of including the site in docs on the main branch is that it will automatically get packaged as part of the releases, so I'd rather not give that up

I think when releases are made we should also produce a `javadoc` jar - which should contain the actual docs... however - I might have misunderstood your comment.

Let's consider for a moment that we would be putting the javadocs into the main branch: in that case I think every commit *must* also update the generated javadocs - to keep them up-to-date ; which would increase commit sizes, burden on contributors - and  could also make commits harder to be reviewed...

> How about if we put the generated javadoc in a branch named "javadoc-3.1". We could point the site to reference that branch on github. Would that be acceptable?

If that's possible - then that would be a great solution!

**Comment by omalley** [comment]
I couldn't make my idea of pushing to an empty branch work. I made such a branch https://github.com/omalley/hive/tree/javadoc-3.1 , but GitHub's presentation logic seems to block my attempts to reference it. Even, https://raw.githubusercontent.com/omalley/hive/javadoc-3.1/index.html serves it up a plain/text.

So I moved all of the references to the subversion. Hopefully, they'll let us keep that svn repository.

It would be good to move forward on this.

**Comment by kgyrtkirk** [comment]
> So I moved all of the references to the subversion. Hopefully, they'll let us keep that svn repository.

awesome :)
+1

**Pull request closed without merging**